presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -2938,6 +2950,20 @@ public static double getHistoryInputTableStatisticsMatchingThreshold(Session ses
         return session.getSystemProperty(HISTORY_INPUT_TABLE_STATISTICS_MATCHING_THRESHOLD, Double.class);
     }
 
+    public static List<PlanCanonicalizationStrategy> getHistoryOptimizationPlanCanonicalizationStrategies(Session session)
+    {
+        List<PlanCanonicalizationStrategy> strategyList;
+        try {
+            strategyList = Splitter.on(",").trimResults().splitToList(session.getSystemProperty(HISTORY_OPTIMIZATION_PLAN_CANONICALIZE_STRATEGY, String.class)).stream()
+                    .map(x -> PlanCanonicalizationStrategy.valueOf(x)).sorted().collect(toImmutableList());
```

Comment:
This is a bit hard to manage.

I think we can remove CONNECTOR strategy(remove safe constants is pretty safe). We can then just use the session property to specify an integer(danger level) - which uses all strategies below that level. Is that sufficient?

Reply:
I took a second look at it, I think it is easier to manage, but less flexible. For example, now we have `ignore_safe_constant` as danger level 1, and `ignore_scan_constant` as danger level 2, if people want to add a new strategy which is between `ignore_safe_constant` and `ignore_scan_constant`, we may need to re-assign the level of `ignore_scan_constant` to 3. This can be problematic as users are now setting the danger level in session property, and it will change behavior of existing settings.
Also it does not allow user to specify the specific strategies to use, but only able to specify a range of strategies.

In order to make it more explicit about the order of the strategies, I added a new field `errorLevel` in the `PlanCanonicalizationStrategy` enum.